cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJobInfo.java

```diff
@@ -46,7 +51,9 @@ public DefaultRuntimeJobInfo(ProgramRunId programRunId, Collection<? extends Loc
         .setProgram(programRunId.getProgram())
         .setRun(programRunId.getRun()).build();
     this.files = Collections.unmodifiableCollection(new ArrayList<>(files));
-    this.jvmProperties = Collections.unmodifiableMap(new LinkedHashMap<>(jvmProperties));
+    this.jvmOpts = jvmOpts;
+    this.jvmProperties = Collections.unmodifiableMap(parseJvmProperties(jvmOpts));
+    LOG.info("JVM properties {}", jvmProperties);
```

Comment:
In general, logging in a constructor is usually a bad idea, tends to result in duplicate logging.

If we want to log, it should be close to where the jvm properties are actually being used

Reply:
reverted this change